packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Expressions/NewInstanceExpression.cs

```diff
@@ -5,20 +5,30 @@
 
 namespace Microsoft.Generator.CSharp.Expressions
 {
-    public sealed record NewInstanceExpression(CSharpType Type, IReadOnlyList<ValueExpression> Parameters, ObjectInitializerExpression? InitExpression = null) : ValueExpression
+    public sealed record NewInstanceExpression(CSharpType? Type, IReadOnlyList<ValueExpression> Parameters, ObjectInitializerExpression? InitExpression = null) : ValueExpression
     {
         private const int SingleLineParameterThreshold = 6;
 
         internal override void Write(CodeWriter writer)
         {
-            writer.Append($"new {Type}");
-            if (Parameters.Count > 0 || InitExpression is not { Parameters.Count: > 0 })
+            writer.Append($"new");
+            if (Type is not null)
+                writer.Append($" {Type}");
+            if (Parameters.Count > 0 || InitExpression is not { Values.Count: > 0 })
```

Comment:
Should we use `Values.Any` instead?

Reply:
typically only do that when we have an IEnumerable and don't know the implementation as to whether it has a `Count` or `Length`.  In this case we know they have a `Count` and can use that property directly.